### PR TITLE
Update build-tools and type tests

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

https://github.com/microsoft/FluidFramework/pull/21070 made some improvements to type tests which were needed to regenerate the type tests for client. This change integrates that, as well as regenerates the type tests with the new generator.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

**Comment by Abe27342** [comment]
ping me when primary reviewer has approved, merge-tree changes are fine.

**Review by alexvy86 — APPROVED** [review]
Per offline discussion, all the already broken types seem to be expected.

For future reference, I think we could handle server separately for type-test update purposes given its separate versioning story. The changes here seem ok to me though.

**Review by Abe27342 — APPROVED** [review]

File: examples/data-objects/table-document/package.json [hunk 1]
```diff
@@ -86,9 +86,9 @@
 		"@biomejs/biome": "^1.6.2",
 		"@fluid-internal/mocha-test-setup": "workspace:~",
 		"@fluid-private/test-version-utils": "workspace:~",
-		"@fluid-tools/build-cli": "^0.38.0",
+		"@fluid-tools/build-cli": "^0.39.0-264124",
```
[the remark this answers is not in the document]
Comment (CraigMacomber) [reply examples/data-objects/table-document/package.json:89]: build cli is used to generate the type tests, so lots of things have it, but maybe this one is unneeded.

Generally example apps have no reason to have type tests,  but I'm not sure whats going on with this package though, as its in @fluid-example, but is not private, has exports, and is not imported in any other examples.

[the remark this answers is not in the document]
Comment (tylerbutler) [reply examples/data-objects/table-document/package.json:89]: Table-document is used by internal partners for historical reasons. It is extremely unique and breaks a lot of "rules" regarding npm scope and supportability. @ChumpChief and @anthony-murphy might know more. There might also be some comments in the fluid-build config file.

**Pull request merged**